### PR TITLE
[DSM] DDP-8384: save filter issue

### PR DESCRIPTION
**Pull request by gmakharat**
## Context

Ticket: https://broadinstitute.atlassian.net/browse/DDP-8384

## Checklist

- [ ] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java [hunk 1]
```diff
@@ -317,6 +319,13 @@ private static ViewFilter getFilterView(@NonNull ResultSet rs, @NonNull Map<Stri
                     }
                     columnList.add(column.substring(column.lastIndexOf(DBConstants.ALIAS_DELIMITER) + 1));
                     columnMap.put(dbElement.getTableAlias(), columnList);
+                } else if (Util.isUnderDsmKey(extractAliasFrom(column))) {
+                    List<Object> columnList = new ArrayList<>();
+                    if (columnMap.containsKey(extractAliasFrom(column))) {
+                        columnList = columnMap.get(extractAliasFrom(column));
+                    }
+                    columnList.add(column.substring(column.lastIndexOf(DBConstants.ALIAS_DELIMITER) + 1));
+                    columnMap.put(column.substring(0, column.indexOf(DBConstants.ALIAS_DELIMITER)), columnList);
```
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java]: So you are adding the column name without the alias?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java [hunk 2]
```diff
@@ -740,6 +762,11 @@ public static ViewFilter parseFilteringQuery(String str, ViewFilter viewFilter)
                             } else if (word.equals(Filter.LIKE)) {
                                 state = 8;
                                 break;
+                            } else if (word.equals(Filter.IS)) {
+                                exact = false;
+                                range = false;
+                                state = 5;
```
Comment (devonbush) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java:780]: what does `state=5`  mean?  please add a comment explaining

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java (hunk 1 above)
Comment (devonbush) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java]: Agree, this needs a comment explaining the intent -- I can't figure it out

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/Util.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.broadinstitute.dsm.model.participant;
+
+import org.broadinstitute.dsm.statics.DBConstants;
+
```
Comment (devonbush) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/Util.java:7]: this code will be cleaner if you make a list of all the possibilities and then use contains for the check.  so
```
public static List<String> DSM_KEY_FIELDS = Arrays.asList(DBConstants.DDP_PARTICIPANT_ALIAS, 
DBConstants.DDP_ONC_HISTORY_DETAIL_ALIAS, DBConstants.DDP_ONC_HISTORY_DETAIL_ALIAS, ...)
```
and then the method can just be
```
public static boolean isUnderDsmKey(String source) {
  return DSM_KEY_FIELDS.contains(source);
}
```

**Review by devonbush — CHANGES_REQUESTED** [review]
This is good progress.  I was able to set up a search for "Consent Audit" not empty, and the participant list displayed correctly. However,  I saved the filter, and it saved, but then trying to load the saved filter resulted in an error 
image

Saving filters with spaces in the name now seems to work -- nice!

**Comment by gmakharat** [comment]
> 

Strange, it works for me, have you switched to frontend branch as well while testing it?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java (hunk 2 above)
Comment (gmakharat) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java:780]: as comment for state 5 says: `//if query contains "IS NOT sth"`

File: pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/model/participant/UtilTest.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        String aliasP = "p";
+        String aliasC = "c";
+        String aliasOd = "oD";
+        boolean allMatch = Stream.of(aliasC, aliasM, aliasOd, aliasP)
+                .allMatch(Util::isUnderDsmKey);
+        assertTrue(allMatch);
+        String noAlias = "noAlias";
+        assertFalse(Util.isUnderDsmKey(noAlias));
+    }
+}
```
Comment (SimoneMaiwald) [inline pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/model/participant/UtilTest.java]: checkstyle will not like this!

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java (hunk 2 above)
Comment (devonbush) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java:780]: where is that comment?  sorry, this is a new part of the codebase for me

**Review by devonbush — APPROVED** [review]

**Pull request merged**